src/services/combineQuery.ts

```diff
@@ -22,5 +23,9 @@ export function combineQuery(parentQuery: Query, childQuery: Query): Query {
     throw new DuplicateParamsError(key)
   }
 
-  return query(`${parentQuery.query}${childQuery.query}`, { ...parentQuery.params, ...childQuery.params })
+  const newQueryString = [parentQuery.query, childQuery.query]
+    .filter(stringHasValue)
```

Comment:
How does this work with an optional param that is undefined vs an empty string?

Reply:
this is inside of `combineQuery`, which is taking a parent route and a child route and creating a new Query that's the combination of both. So there is no undefined or empty strings here